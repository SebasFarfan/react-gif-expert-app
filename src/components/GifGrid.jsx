import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

  const {images, isLoading} = useFetchGifs(category);
  // console.log({images, isLoading});
  

  return (
    <>
      <h3>{category}</h3>
      {/* <p>Hola Mundo</p> */}
      {
        isLoading && (<h2>Cargando....</h2>)
      }
      <div className="card-grid">
        {/* {
          images.map(image =>(
            <li key={image.id}>{image.title}</li>

          ))
        } */}
        {images.map((image) => (
          <GifItem key={image.id} {...image}/>
        ))}
      </div>
    </>
  );
};
