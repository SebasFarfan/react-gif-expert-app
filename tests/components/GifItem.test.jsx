import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas sobre <GifItem />", () => {
    const title = 'dragon ball';
    const url = 'https://dragon-ball.com/';
  test("Debe hacer match con el snapshot", () => {
    const {container} = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe Mostrar la imagen con la URL y el ALT indicado', () => { 
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    const {src,alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
   })
});
