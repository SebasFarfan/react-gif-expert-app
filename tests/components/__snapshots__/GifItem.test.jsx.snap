// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="dragon ball"
      src="https://dragon-ball.com/"
    />
    <p>
      dragon ball
    </p>
  </div>
</div>
`;
